x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -445,30 +454,32 @@ export class EnterpriseSearchPlugin implements Plugin {
       title: ANALYTICS_PLUGIN.NAME,
     });
 
-    core.application.register({
-      appRoute: INFERENCE_ENDPOINTS_PLUGIN.URL,
-      category: DEFAULT_APP_CATEGORIES.enterpriseSearch,
-      deepLinks: relevanceLinks,
-      euiIconType: INFERENCE_ENDPOINTS_PLUGIN.LOGO,
-      id: INFERENCE_ENDPOINTS_PLUGIN.ID,
-      mount: async (params: AppMountParameters) => {
-        const kibanaDeps = await this.getKibanaDeps(core, params, cloud);
-        const { chrome, http } = kibanaDeps.core;
-        chrome.docTitle.change(INFERENCE_ENDPOINTS_PLUGIN.NAME);
+    if (isEnterpriseLicense) {
```

Comment:
this is creating a race condition: the subscription above does need to have been resolved at the time this application registration is called. Instead, this registration should be called inside that subscription. That way you also don't need to track the isEnterpriseLicense variable outside of that subscription.

Reply:
Updated